src/Fable.Transforms/FSharp2Fable.fs

```diff
@@ -171,15 +171,20 @@ let private transformTraitCall com (ctx: Context) r typ (sourceTypes: FSharpType
 let private transformObjExpr (com: IFableCompiler) (ctx: Context) (objType: FSharpType)
                     baseCallExpr (overrides: FSharpObjectExprOverride list) otherOverrides =
 
-    let mapOverride (over: FSharpObjectExprOverride) =
+    let mapOverride (typ: FSharpType) (over: FSharpObjectExprOverride) =
       trampoline {
         let ctx, args = bindMemberArgs com ctx over.CurriedParameterGroups
         let! body = transformExpr com ctx over.Body
         let value = Fable.Function(Fable.Delegate args, body, None)
         let name, kind =
             match over.Signature.Name with
             | Naming.StartsWith "get_" name when countNonCurriedParamsForOverride over = 0 ->
-                name, Fable.ObjectGetter
+                // performance optimization, compile get_Current as instance call instead of a getter
+                if over.Signature.Name = "get_Current" &&
+                    (typ.TypeDefinition.FullName = "System.Collections.Generic.IEnumerator`1" ||
+                     typ.TypeDefinition.FullName = "System.Collections.IEnumerator")
+                then name, Fable.ObjectMethod false
+                else name, Fable.ObjectGetter
```

Comment:
Instead of adding ad-hoc exceptions, we can get this "for free" in Fable 3 by removing `Enumerator` from the list of non-mangled System interfaces here (though in that case we need to change the TypeScript code that uses it): https://github.com/fable-compiler/Fable/blob/e84aa4d58bb43447a3593e7a7f9683db37192a38/src/Fable.Transforms/FSharp2Fable.Util.fs#L1030-L1046

Reply:
@alfonsogarciacaro I'm not sure we can do that reliably without replicating the name mangling in TS. Also, this PR targets v.current, not v.next. Anyway, this is still WIP, since the perf in this PR is still below the original Types.List.